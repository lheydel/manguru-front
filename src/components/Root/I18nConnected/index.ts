export * from './I18nConnected';
