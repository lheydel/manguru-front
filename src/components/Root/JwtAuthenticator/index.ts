import { JwtAuthenticator } from './JwtAuthenticator';
export default JwtAuthenticator;
export * from './JwtAuthenticator';
