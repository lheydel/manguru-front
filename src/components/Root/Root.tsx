import React from 'react';
import { Provider } from 'react-redux';
import { Store } from 'redux';
import { ConnectedRouter } from 'connected-react-router';
import { History } from 'history';
import { Routes } from '../../routes';
import { I18nConnected } from './I18nConnected';

export interface RootProps {
  store: Store;
  history: History;
}

export const Root: React.FC<RootProps> = ({store, history}: RootProps) => (
    <Provider store={store}>
      <I18nConnected>
        <ConnectedRouter history={history}>
          <Routes />
        </ConnectedRouter>
      </I18nConnected>
    </Provider>
);
