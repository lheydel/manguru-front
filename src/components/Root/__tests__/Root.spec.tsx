import React from 'react';
import { RootProps, Root } from '../Root';
import { shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import toJson from 'enzyme-to-json';
import createMockStore from 'redux-mock-store';

const mockStore = createMockStore();

describe('Root component', () => {
    const props: RootProps = {
        store: mockStore(),
        history: createBrowserHistory()
    };

    it('should renders itself and children', () => {
        const wrapper = shallow(<Root {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
