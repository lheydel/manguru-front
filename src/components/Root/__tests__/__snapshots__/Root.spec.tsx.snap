// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root component should renders itself and children 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CookiesProvider
    cookies={
      Cookies {
        "HAS_DOCUMENT_COOKIE": false,
        "changeListeners": Array [],
        "cookies": Object {},
      }
    }
  >
    <Connect(RawI18nConnected)>
      <Connect(ConnectedRouterWithContext)
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <withCookies(Connect(RawJwtAuthenticator))>
          <Routes />
        </withCookies(Connect(RawJwtAuthenticator))>
      </Connect(ConnectedRouterWithContext)>
    </Connect(RawI18nConnected)>
  </CookiesProvider>
</Provider>
`;
